src/popup/popup.js

```diff
@@ -288,25 +293,34 @@ $('#choose-license').comboTree({
 });
 loadUserDefaults();
 
-function loadStoredSearch() {
-  if (localStorage !== null) {
-    inputText = localStorage.getItem('title');
-    elements.inputField.value = inputText;
-
-    pageNo = 1;
-    if (localStorage.getItem(pageNo)) {
-      removeNode('primary__initial-info');
-      const pageData = Object.values(JSON.parse(localStorage.getItem(pageNo)));
-      addThumbnailsToDOM(pageData);
-      pageNo = Number(pageNo) + 1;
+async function loadStoredSearch() {
+  await chrome.storage.sync.get(['enableSearchStorage'], (res) => {
+    if (res.enableSearchStorage === undefined) {
+      enableSearchStorageOption = true;
+      // eslint-disable-next-line quote-props
+      chrome.storage.sync.set({ 'enableSearchStorage': enableSearchStorageOption });
+    } else enableSearchStorageOption = res.enableSearchStorage;
```

Comment:
This is logically different from the purpose of `loadStoredSearch()` function. Move this to a different one with name like `setEnableSearchStorageOptionVariable()` and call it inside `loadStoredSearch()`.